### PR TITLE
KAFKA-3131: enable error level for SSLException logs

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@ijuma @rajinisivaram This is a minor change to enable error logs for SSL exceptions. most of the times, users misconfigure SSL and don't get any clue about underlying errors. error logs may help in those scenarios. let me know your thoughts on this.

File: clients/src/main/java/org/apache/kafka/common/network/Selector.java [hunk 1]
```diff
@@ -463,17 +465,23 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
...
                 close(channel, true);
             } finally {
                 maybeRecordTimePerConnection(channel, channelStartTimeNanos);
             }
         }
     }
 
+    private void logException(Exception e, String desc) {
+        if (e instanceof SSLException)
+            log.error("Connection with {} disconnected", desc, e);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/network/Selector.java]: Are all `SSLExceptions` fatal? Also, we should probably update the log message.

Comment (omkreddy) [reply clients/src/main/java/org/apache/kafka/common/network/Selector.java]: AFAIK, SSLHandshakeException, SSLKeyException, SSLPeerUnverifiedException are fatal exceptions, SSLProtocolException is not common.  We can check with @rajinisivaram  

Updated logs.

Comment (rajinisivaram) [reply clients/src/main/java/org/apache/kafka/common/network/Selector.java]: On the broker-side this is not fatal, but typically caused by a mis-configured client. For clients, it is typically fatal, but could sometimes just be a clock-mismatch where a retry could succeed.

Comment (omkreddy) [reply clients/src/main/java/org/apache/kafka/common/network/Selector.java]: @ijuma @rajinisivaram  any suggestion on this PR?  also Is there is a way to identify broker-side usage with the client-side usage of Selector class?

**Comment by omkreddy** [comment]
Closing this PR in favor of KAFKA-5920 and related changes

**Pull request closed without merging**